### ISSUE TITLE
Add the feture of Video Filters(rules to filter out videos which has a specified text in its title) 

### ISSUE THREAD

**Issue by 2000prasanth** [TeamNewPipe/NewPipe#10573]
### Checklist

- [X] I made sure that there are *no existing issues* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have read the [FAQ](https://newpipe.net/FAQ/) and my problem isn't listed.
- [X] I'm aware that this is a request for NewPipe itself and that requests for adding a new service need to be made at [NewPipeExtractor](https://github.com/TeamNewPipe/NewPipeExtractor/issues).
- [X] I have taken the time to fill in all the required details. I understand that the feature request will be dismissed otherwise.
- [X] This issue contains only one feature request.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### Feature description

There is a feture in clipious another youtube client that basically filters out videos that has a certain text in its title or description. 

### Why do you want this feature?

![Screenshot_20231114-191631~2](https://github.com/TeamNewPipe/NewPipe/assets/60725771/40bca45e-5396-4648-aa2b-8808df8816d0)
Basically the user adds any text inorder to filter out videos that contain that text, it works better than YouTube restricted mode
